apps/aefate/src/aefa_chain_api.erl

```diff
@@ -194,9 +201,41 @@ traverse_to_key_hash(H, KeyHash) ->
         _Height -> traverse_to_key_hash(H, aec_headers:prev_key_hash(Header))
     end.
 
--spec contract_fate_code(pubkey(), state()) -> 'error' |
-                                               {'ok', {term(), aect_contracts:vm_version()}, state()}.
-contract_fate_code(Pubkey, #state{primop_state = PState} = S) ->
+-spec put_contract(aect_contracts:contract(), state()) -> state().
+put_contract(Contract, #state{primop_state = PS} = S) ->
+    PK = aect_contracts:pubkey(Contract),
+    {Payable, PS0} = case aect_contracts:code(Contract) of
+                  {code, SerCode} ->
+                      #{payable := P} = aeser_contract_code:deserialize(SerCode),
+                      {P, PS};
+                  {ref, Clonee} ->
+                      case aeprimop_state:find_account(aeser_id:specialize(Clonee, contract), PS) of
+                          none -> error;
```

Comment:
Is it OK that you hard crash with a match error in that case?

Reply:
This is an internal error which happens only in case someone refers to a contract without an account which should be invalid in the first place. The match crash shall point to a code that clearly shows what was the issue